### ISSUE TITLE
autofix for UP035

### ISSUE THREAD

**Issue by Olegt0rr** [astral-sh/ruff#7147]
Source:

```python
from typing import Dict, List


def get_values(some_dict: Dict) -> List:
    return list(some_dict.values())
```

Command:
```
ruff check --select UP draft.py --fix
```

Fact result:
```python
from typing import Dict, List


def get_values(some_dict: dict) -> list:
    return list(some_dict.values())
```
+ 
```
draft.py:1:1: UP035 `typing.Dict` is deprecated, use `dict` instead
draft.py:1:1: UP035 `typing.List` is deprecated, use `list` instead
```

Expected result:
```python
def get_values(some_dict: dict) -> list:
    return list(some_dict.values())

```

it seems we should just remove import string if there's no usage

**Comment by Olegt0rr** [comment]
Why it's bad to remove `Dict` import for just replaced `Dict` with `dict`?
It seems single UP035 should be able to solve problem without any additional rules.

imho

**Comment by charliermarsh** [comment]
I agree that it would probably be nice to remove them (though _arguably_ unexpected), but it's more a design choice on our side: by keeping each rule's autofix focused on just the diagnostic it enforces, and running diagnostics repeatedly (i.e., if you run with `--select UP --select F`, we'll first remove the `Dict` usages, and then in a second iteration, remove the `Dict` imports), it really simplifies the number of concerns that each rule needs to have. I think this is unlikely to change in the near future, though perhaps we'll end up special-casing unused imports at some point.
